pyls/plugins/pylint_lint.py

```diff
@@ -0,0 +1,140 @@
+# Copyright 2018 Google LLC.
+"""Linter plugin for pylint."""
+import collections
+import json
+import logging
+import sys
+
+from pylint.epylint import py_run
+from pyls import hookimpl, lsp
+
+
+log = logging.getLogger(__name__)
+
+
+class PylintLinter(object):
+    last_diags = collections.defaultdict(list)
+
+    @classmethod
+    def lint(cls, document, is_saved, flags=''):
+        """Plugin interface to pyls linter.
+
+        Args:
+            document: The document to be linted.
+            is_saved: Whether or not the file has been saved to disk.
+            flags: Additional flags to pass to pylint. Not exposed to
+                pyls_lint, but used for testing.
+
+        Returns:
+            A list of dicts with the following format:
+
+                {
+                    'source': 'pylint',
+                    'range': {
+                        'start': {
+                            'line': start_line,
+                            'character': start_column,
+                        },
+                        'end': {
+                            'line': end_line,
+                            'character': end_column,
+                        },
+                    }
+                    'message': msg,
+                    'severity': lsp.DiagnosticSeverity.*,
+                }
+        """
+        log.debug('Running pylint on %s', document.path)
+        if not is_saved:
+            # Pylint can only be run on files that have been saved to disk.
+            # Rather than return nothing, return the previous list of
+            # diagnostics. If we return an empty list, any diagnostics we'd
+            # previously shown will be cleared until the next save. Instead,
+            # continue showing (possibly stale) diagnostics until the next
+            # save.
+            log.debug('Cannot run pylint, file has not been saved to disk.')
```

Comment:
This is going to log a lot, can we remove?

Reply:
Is debug log level normal? Easy enough to remove, but it can be helpful in debugging and I assume that's not a commonly used log level.